### DISCUSSION TITLE
How can I tell JavaCPP to ignore this macro?

### DISCUSSION THREAD

**Discussion by nyholku** [bytedeco/javacpp#550] [Q&A] [answered]
OCC has a nasty way of adding code to a class like with a macro DEFINE_STANDARD_ALLOC like this:
```
class Standard_Transient
{
public:
  // Standard OCCT memory allocation stuff
  DEFINE_STANDARD_ALLOC

public:

  //! Empty constructor
  Standard_Transient() : count(0) {}
```

I've tried to skip this, but I can figure out a regexp that would not also skip everything starting from next `public:` also.

```
infoMap.put(new Info("Standard_Transient.hxx").linePatterns("public:","\\s+DEFINE_STANDARD_ALLOC").skip());
``

Ideally I would like to replace the DEFINE_STANDARD_ALLOC a null C++ class member like `void somename(){}` but skipping those lines would be enough.

**Comment by saudet** [comment, marked as answer]
We can redefine macros as explained here:
https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#redefining-the-code-of-a-macro
In this case, something like this should work:
```java
infoMap.put(new Info("DEFINE_STANDARD_ALLOC").cppText("#define DEFINE_STANDARD_ALLOC void somename(){}"));
```

**Reply by nyholku** [reply]
Thanks. I've seen and tried that before. Did not work.

If I do :

```
infoMap.put(new Info("DEFINE_STANDARD_ALLOC").cppText("#define DEFINE_STANDARD_ALLOC void somename(){}"));
```

Then javacpp will try to generate call to `somename` and that of course fails. Addings `.skip()` to above has no effect.

The root problem is that because of the double `public:` in the header file. I cannot leave the macro empty because then we have `public:public:` which the parser does not like (not sure if it is valid C++ either). So I was searching for something that would ignore everthing from the first `public:` to the `DEFINE_STANDARD_ALLOC` inclusive, or some harmless but valid C++ declaration to take its place.

**Reply by saudet** [reply]
If you just want to have it ignored, erase the right part of the definition like this:
```java
infoMap.put(new Info("DEFINE_STANDARD_ALLOC").cppText("#define DEFINE_STANDARD_ALLOC"));
```
I'm sure we can have `public:public:`, that's fine.

**Reply by nyholku** [reply]
Thanks, that did the trick! I was pretty sure I tried that before but apparently there was something else wrong at that time.
